AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/MutationQueue/OutgoingMutationQueueTestsWithMockStateMachine.swift

```diff
@@ -65,33 +65,58 @@ class OutgoingMutationQueueMockStateTest: XCTestCase {
         waitForExpectations(timeout: 1)
     }
 
-    func testRequestingEvent_subscriptionSetup() {
+    func testRequestingEvent_subscriptionSetup() throws {
         let semaphore = DispatchSemaphore(value: 0)
         stateMachine.pushExpectActionCriteria { action in
             XCTAssertEqual(action, OutgoingMutationQueue.Action.receivedSubscription)
             semaphore.signal()
         }
         stateMachine.state = .starting(apiBehavior, publisher)
         semaphore.wait()
+
+        let json = "{\"id\":\"1234\",\"title\":\"t\",\"content\":\"c\",\"createdAt\":\"2020-09-03T22:55:13.424Z\"}"
         let futureResult = MutationEvent(modelId: "1",
                                          modelName: "Post",
-                                         json: "{}",
+                                         json: json,
                                          mutationType: MutationEvent.MutationType.create)
         eventSource.pushMutationEvent(futureResult: .success(futureResult))
 
         let enqueueEvent = expectation(description: "state requestingEvent, enqueueEvent")
         let processEvent = expectation(description: "state requestingEvent, processedEvent")
+
         stateMachine.pushExpectActionCriteria { action in
             XCTAssertEqual(action, OutgoingMutationQueue.Action.enqueuedEvent)
             enqueueEvent.fulfill()
         }
+
+        let mutateAPICallExpecation = expectation(description: "Call to api category for mutate")
+        var listenerFromRequest: GraphQLOperation<MutationSync<AnyModel>>.ResultListener!
+        let responder = MutateRequestListenerResponder<MutationSync<AnyModel>> { _, eventListener in
```

Comment:
I updated the logic of this test a bit because the test is using `MockAPIPlugin` and I want it to return a response so that the `outboxMutationProcessed` will be fired, received and fulfilled